spring-kafka/src/main/java/org/springframework/kafka/listener/PartitionPausingBackOffManagerFactory.java

```diff
@@ -161,7 +173,20 @@ private KafkaConsumerTimingAdjuster getOrCreateBackOffTimingAdjustmentManager()
 		if (this.timingAdjustmentManager != null) {
 			return this.timingAdjustmentManager;
 		}
-		return new WakingKafkaConsumerTimingAdjuster(this.taskExecutor);
+		Assert.isTrue(this.taskExecutor == null || this.maxThreadPoolSize == null,
+				() -> "maxThreadPoolSize should not be used when providing a TaskExecutor instance" +
+						"- configure the TaskExecutor instance directly instead.");
+		return this.taskExecutor != null
+				? new WakingKafkaConsumerTimingAdjuster(this.taskExecutor)
+				: new WakingKafkaConsumerTimingAdjuster(createTaskExecutor());
+	}
+
+	private TaskExecutor createTaskExecutor() {
+		ThreadPoolTaskExecutor taskExecutor = new ThreadPoolTaskExecutor();
+		JavaUtils.INSTANCE.acceptIfNotNull(this.maxThreadPoolSize, taskExecutor::setMaxPoolSize);
+		taskExecutor.initialize();
+		super.addApplicationListener((ApplicationListener<ContextClosedEvent>) event -> taskExecutor.shutdown());
```

Comment:
I still in doubts for all of this stuff around internal executor...
Can you, please, bring the bean back and let's revise it in a separate issue?

Reply:
Sure, no worries, I'll bring it back that and open the issue. Thanks.

Reply:
I've moved it back - how grateful I am for `git` in these moments 😄

Also opened the issue: https://github.com/spring-projects/spring-kafka/issues/2239

Please let me know if there's anything else.

Thanks.